src/main/java/com/synopsys/integration/alert/workflow/scheduled/update/UpdateChecker.java

```diff
@@ -112,31 +128,83 @@ private boolean isComprisedOfNumericTokens(final String version) {
                    .allMatch(NumberUtils::isParsable);
     }
 
-    private Comparator<String> versionOrder() {
-        return (firstVersion, secondVersion) -> {
-            final String[] firstVersionTokens = StringUtils.split(firstVersion, VERSION_SEPARATOR);
-            final String[] secondVersionTokens = StringUtils.split(secondVersion, VERSION_SEPARATOR);
+    private Comparator<DockerTagModel> tagOrder() {
+        return (firstTag, secondTag) -> compareVersionAndDates(firstTag.getName(), firstTag.getLastUpdated(), secondTag.getName(), secondTag.getLastUpdated(), this::compareDateStrings);
+    }
 
-            for (int i = 0; i < firstVersionTokens.length && i < secondVersionTokens.length; i++) {
-                final int firstToken = Integer.parseInt(firstVersionTokens[i]);
-                final int secondToken = Integer.parseInt(secondVersionTokens[i]);
+    private int compareVersionAndDates(final String firstVersion, final String firstDate, final String secondVersion, final String secondDate, final BiFunction<String, String, Integer> dateComparison) {
```

Comment:
Why do we need to pass a method reference here? Do we have more than one date comparison method? These methods are both in the same class.